### PR TITLE
Refactor project build and run

### PR DESCRIPTION
**Pull request by JohnNiang**
This PR mainly adapts PR <https://github.com/halo-dev/halo/pull/5314>.

```release-note
None
```

### CODE REVIEW

File: docs/developer-guide/core/build.md [hunk 1]
```diff
@@ -83,18 +59,19 @@ cd path/to/halo
 ./gradlew clean build -x check
 ```
 
-构建完成之后，在 halo 项目下产生的 `application/build/libs/application-$version.jar` 即为构建完成的文件。
+构建完成之后，在 Halo 项目下产生的 `application/build/libs/halo-${version}.jar` 即为构建完成的文件。
```
Comment (ruibaby) [inline docs/developer-guide/core/build.md:62]: 建议在这行下面添加 jar 部署方式的文档提示。

https://docs.halo.run/getting-started/install/jar-file

File: docs/developer-guide/core/build.md [hunk 2]
```diff
@@ -83,18 +59,19 @@ cd path/to/halo
...
-在进行之前，请确保已经完成上述操作，最终需要确认在 halo 项目的 `application/build/libs/` 目录已经包含了 `application-$version.jar` 文件。
+在此之前，请确认已经构建好了 Fat Jar。
 
 ```bash
 cd path/to/halo
 ```
 
 ```bash
-docker build -t halo-dev/halo:2.12.0 .
+# 请替换成 ${tag_name}
```
Comment (ruibaby) [inline docs/developer-guide/core/build.md:73]: 应该是请替换 ${tag_name}？

**Review by ruibaby — CHANGES_REQUESTED** [review]

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Review by guqing — APPROVED** [review]

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**